src/imitation/algorithms/dagger.py

```diff
@@ -75,65 +75,88 @@ def reconstruct_trainer(
       trainer: a reconstructed `DAggerTrainer` with the same state as the
         previously-saved one.
     """
-    checkpoint_path = os.path.join(scratch_dir, "checkpoint-latest.pt")
+    checkpoint_path = pathlib.Path(scratch_dir, "checkpoint-latest.pt")
     return th.load(checkpoint_path, map_location=utils.get_device(device))
 
 
-def _save_trajectory(
-    npz_path: str,
+def _save_dagger_demo(
```

Comment:
I probably wrote the original version of this. I don't know why I did it this way, and can't think of any obvious objection to changing it.

Reply:
To be fair, I think NPZ is usually more efficient than pickle for large NumPy arrays. Although the most efficient thing on disk would probably be to flatten the trajectories into transitions and then reconstruct them from dones, since the overhead of lots of small NumPy arrays is significant. This is probably premature optimization for now though.

Reply:
I don't have the bandwidth to refactor this right now I think, but I'll flag this as ~"possibly redundant with `types.save`, though saving to npz is more efficient than what we do there"